Do-It/Controller/DoItTableViewController.swift

```diff
@@ -101,24 +123,75 @@ class DoItTableViewController: UIViewController, UITableViewDelegate, UITableVie
         // self.performSegue(withIdentifier: "yourIdentifier", sender: self)
     }
 
+    func tableView(_ tableView: UITableView,
+                   commit editingStyle: UITableViewCell.EditingStyle,
+                   forRowAt indexPath: IndexPath) {
+        if editingStyle == .delete {
+            doIts.remove(at: indexPath.row)
+            filteredDoIts = doIts
+            tableView.deleteRows(at: [indexPath], with: .fade)
+        }
+        //        } else if editingStyle == .insert {
+        //            // Create a new instance of the appropriate class,
+        //        // insert it into the array, and add a new row to the table view.
+        //        }
+
+        tableView.reloadData()
+    }
 }
 
-// Methods called when buttons pressed
+// MARK: - Methods called when buttons pressed
 extension DoItTableViewController {
 
-    @IBAction func editButtonPressed() {
-        // IMPLEMENT ME
-        print("edit button pressed")
+    @objc func editButtonPressed() {
+
+        if navigationEditItem.rightBarButtonItem?.title == "Cancel" {
+            navigationEditItem.rightBarButtonItem?.title = "Edit"
+            navigationEditItem.title = "Do-Its"
+            navigationEditItem.titleView = nil
+        } else {
+            tableView.setEditing(!tableView.isEditing, animated: true)
+            navigationEditItem.rightBarButtonItem?.title = tableView.isEditing ? "Done" : "Edit"
+        }
+
     }
 
     @IBAction func searchButtonPressed() {
         // IMPLEMENT ME
+        // present search bar
+        // change
         print("search button pressed")
+
+        navigationEditItem.titleView = searchBar
+        navigationEditItem.title = ""
+        navigationEditItem.rightBarButtonItem?.title = "Cancel"
     }
 
     @IBAction func composeButtonPressed() {
         // IMPLEMENT ME
+        // go to addDoItViewController
         print("compose button pressed")
     }
 
 }
+
+// Mark: - Search Bar Delegate
+extension DoItTableViewController {
+    func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
```

Comment:
Let's pull searching logic out into a separate class.
It could probably even implement the `FilterSpecification` protocol that Jett wrote.